x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/InternalInferModelAction.java

```diff
@@ -24,12 +24,12 @@
 import java.util.Map;
 import java.util.Objects;
 
-public class InferModelAction extends ActionType<InferModelAction.Response> {
+public class InternalInferModelAction extends ActionType<InternalInferModelAction.Response> {
 
-    public static final InferModelAction INSTANCE = new InferModelAction();
+    public static final InternalInferModelAction INSTANCE = new InternalInferModelAction();
     public static final String NAME = "cluster:admin/xpack/ml/inference/infer";
```

Comment:
If this action will now only ever be allowed to be called by other internal components it is probably worth changing `admin` to `internal` in this action name.  See `KillProcessAction` for an example.

But actually I think it would be nice if there was an inference endpoint similar to this that _was_ publicly accessible by users with some ML cluster privilege but obviously didn't allow overriding of the licensing.  (That's one for a different PR though.)

Reply:
@droberts195 100% agree, that is why I renamed it, but I forgot to rename the action name. Will do it now so I don't forget!